torch/_dynamo/guards.py

```diff
@@ -2524,6 +2541,29 @@ def _unpickle_dispatch_key_set(cls, raw_repr: int):
 
     def reducer_override(self, obj):
         if isinstance(obj, torch.Tensor) and obj.device.type != "meta":
+            from torch.utils._python_dispatch import is_traceable_wrapper_subclass
+
+            if is_traceable_wrapper_subclass(obj):
+                # inner_data is a list of tuples of:
+                #   (inner attr name, unpickle func, tuple of func inputs)
+                # This supports traceable wrapper subclass inner tensors.
+                inner_data = []
+                attrs, ctx = obj.__tensor_flatten__()
```

Comment:
Not very familiar with the contract here. I guess we enforce __tensor_flatten__ to be always defined with tensor subclasses? (otherwise we can fire a user friendly error message to remind them to do so)

Reply:
Yep that's right - both `__tensor_flatten__()` and `__tensor_unflatten__()` must be defined to be considered a "traceable wrapper subclass". Such tensors must be decomposable into a set of inner tensors.